stixcore/products/level0/housekeeping.py

```diff
@@ -0,0 +1,204 @@
+        data['watchdog_state'] = packets.get_value('NIXD0169', attr='value')
+        data['received_spw_packets'] = packets.get_value('NIXD0079')
+        data['rejected_spw_packets'] = packets.get_value('NIXD0078')
+        data['hk_dpu_1v5_v'] = packets.get_value('NIXD0035')
+        data['hk_ref_2v5_v'] = packets.get_value('NIXD0036')
+        data['hk_dpu_2v9_v'] = packets.get_value('NIXD0037')
+        data['hk_psu_temp_t'] = packets.get_value('NIXD0024')
+        data['fdir_status'] = packets.get_value('NIX00085')
+        data['fdir_status_mask_of_hk_temperature'] = packets.get_value('NIX00161')
+        data['fdir_status_mask_of_hk_voltage'] = packets.get_value('NIX00162')
+        data['hk_selftest_status_flag'] = packets.get_value('NIXD0163', attr='value')
+        data['memory_status_flag'] = packets.get_value('NIXD0164', attr='value')
+        data['fdir_status_mask_of_hk_current'] = packets.get_value('NIXD0165')
+        data['number_executed_tc'] = packets.get_value('NIX00166', attr='value')
+        data['number_sent_tm'] = packets.get_value('NIX00167', attr='value')
+        data['number_failed_tm_gen'] = packets.get_value('NIX00168', attr='value')
+        data['control_index'] = range(len(control))
+
+        return cls(service_type=service_type, service_subtype=service_subtype, ssid=ssid,
+                   control=control, data=data)
+
+    @classmethod
+    def is_datasource_for(cls, *, service_type, service_subtype, ssid, **kwargs):
+        return (kwargs['level'] == 'L0' and service_type == 3
+                and service_subtype == 25 and ssid == 1)
+
+
+class MaxiReport(QLProduct):
+    """
+    Maxi house keeping reported in all modes while the flight software is running.
+    """
+    def __init__(self, *, service_type, service_subtype, ssid, control, data, **kwargs):
+        super().__init__(service_type=service_type, service_subtype=service_subtype,
+                         ssid=ssid, control=control, data=data, **kwargs)
+        self.name = 'maxi'
+        self.level = 'L0'
+        self.type = 'hk'
+
+    @classmethod
+    def from_levelb(cls, levelb):
+        packets = super().from_levelb(levelb)
+
+        service_type = packets.get('service_type')[0]
+        service_subtype = packets.get('service_subtype')[0]
+        ssid = packets.get('pi1_val')[0]
+
+        control = Control()
+        control['scet_coarse'] = packets.get('scet_coarse')
+        control['scet_fine'] = packets.get('scet_fine')
+        control['integration_time'] = 0
+        control['index'] = range(len(control))
+
+        # Create array of times as dt from date_obs
+        times = [DateTime(ct, ft).as_float() for ct, ft in control['scet_coarse', 'scet_fine']]
+
+        # Data
+        data = Data()
+        data['time'] = times
+        data['sw_running'] = packets.get_value('NIXD0021', attr='value')
+        data['instrument_number'] = packets.get_value('NIXD0022', attr='value')
```

Comment:
Could probably replace this and the similar block above with a loop using a modified version of the the parameters name from the IDB? Override one that need to be modified or fix in idb

Reply:
hm yes that might be a good sugestion